### PR TITLE
tools: remove perl in mtx changer

### PR DESCRIPTION
**Pull request by bruno-at-bareos**
This PR aims to cleanup our mtx-changer script.
We removed the perl dependency making it for friendly for containerized environment.
Some efforts have been put into reformatting sources and simplifying expression and usage.
 
- additionally we request python3 in webui selenium testing

#### Please check

- [X] Short description and the purpose of this PR is present _above this paragraph_
- [X] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)
- [Boy Scout Rule](https://docs.bareos.org/DeveloperGuide/generaldevel.html#boy-scout-rule)

### Checklist for the _reviewer_ of the PR (will be processed by the Bareos team)
Make sure you check/merge the PR using `devtools/pr-tool` to have some simple automated checks run and a proper changelog record added.

##### General
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Check backport line
- [x] Required backport PRs have been created

##### Source code quality
- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR

##### Tests
- [x] Decision taken that a test is required (if not, then remove this paragraph)
- [x] The choice of the type of test (unit test or systemtest) is reasonable
- [x] Testname matches exactly what is being tested
- [x] On a fail, output of the test leads quickly to the origin of the fault

### CODE REVIEW

**Comment by arogge** [comment]
I'm going to write a super-simple mockup-mtx and a systemtest to make sure this doesn't break.

File: core/scripts/mtx-changer.in [hunk 1]
```diff
@@ -131,166 +132,223 @@ wait_for_drive() {
...
+    #cat ${TMPFILE} | grep "Data Transfer Element" | awk "{print \"D:\"\$4 \$7 \$9 \$10}" | sed "s/=/:/" | sed "s/Full/F:/" | sed "s/Empty/E/"
+    awk_code=$(
+cat << '_EOT_'
+    /^Data Transfer Element .*[0-9]:/ {
+      gsub("Data Transfer Element ","",$1)
+      gsub("Full .*",":F",$2)
+      gsub("Empty",":E",$2)
+      gsub("VolumeTag = ",":",$3)
+      print "D:" $1 $2 $3
+    }
```
Comment (arogge) [inline core/scripts/mtx-changer.in:294]: we still have a bug in here. The output for a drive with a volume loaded should look like this:
`D:<drive>:F:<slot-currently-loaded>:<volume-tag>`
e.g.
`D:0:F:3:000003L8` 
for volume 000003L8 loaded from slot 3 into drive 0

**Review by arogge — CHANGES_REQUESTED** [review]
this looks pretty good. There's just one little bug the systemtest I added found.

File: core/scripts/mtx-changer.in (hunk 1 above)
Comment (bruno-at-bareos) [reply core/scripts/mtx-changer.in]: Ok seen, I've fixed that now with gensub

File: systemtests/tests/mtx-changer/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+#   Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with this program; if not, write to the Free Software
+#   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+#   02110-1301, USA.
+
+get_filename_component(BASENAME ${CMAKE_CURRENT_BINARY_DIR} NAME)
+create_systemtest(${SYSTEMTEST_PREFIX} ${BASENAME})
```
Comment (sebsura) [inline systemtests/tests/mtx-changer/CMakeLists.txt:21]: The test probably needs to lock the `autochanger_resource` right ? Or can this be run concurrently with the other mtx-tests ?  See e.g. `block-size/CMakeLists.txt`, `autochanger/CMakeLists.txt`.

Comment (bruno-at-bareos) [reply systemtests/tests/mtx-changer/CMakeLists.txt:21]: Well the test is a mock, so it doesn't really need any real resource. The new tool read output of pre-recorded mtx command, and the new output should be equivalent to the pre-existing values.

File: core/scripts/mtx-changer.in [hunk 3]
```diff
@@ -131,166 +132,223 @@ wait_for_drive() {
...
+cat << '_EOT_'
+    /^ .*Storage Element .*[0-9]:.*Full/ {
+      gsub("[ ].*Storage Element ","",$1)
+      gsub("VolumeTag=","",$3)
+      gsub(/ *$/, "", $3)
+      print $1 ":" $3
+    }
+_EOT_
+    )
+    awk -F ":" "$awk_code" "${TMPFILE}"
```
Comment (arogge) [inline core/scripts/mtx-changer.in:239]: I just realized that this is way more complicated than needed.
We don't need to assign that to a variable, but can just make `awk` read the script from stdin like this:
```suggestion
    awk -f - -F ":" "${TMPFILE}" << '_EOT_'
      /^ .*Storage Element .*[0-9]:.*Full/ {
        gsub("[ ].*Storage Element ","",$1)
        gsub("VolumeTag=","",$3)
        gsub(/ *$/, "", $3)
        print $1 ":" $3
      }
_EOT_
```
That should work for all the other `awk` calls, too.

Comment (bruno-at-bareos) [reply core/scripts/mtx-changer.in]: done

**Review by arogge — APPROVED** [review]
Of course, there is always a lot more stuff to optimize, but that's probably true for every shell-script we have.
This is really a huge improvement! Thank you!

**Pull request merged**